src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/service-daemon-list/service-daemon-list.component.ts

```diff
@@ -68,10 +80,45 @@ export class ServiceDaemonListComponent implements OnInit, OnChanges, AfterViewI
     private hostService: HostService,
     private cephServiceService: CephServiceService,
     private orchService: OrchestratorService,
-    private relativeDatePipe: RelativeDatePipe
+    private relativeDatePipe: RelativeDatePipe,
+    public actionLabels: ActionLabelsI18n,
+    private authStorageService: AuthStorageService,
+    private daemonService: DaemonService,
+    private notificationService: NotificationService
   ) {}
 
   ngOnInit() {
+    this.permissions = this.authStorageService.getPermissions();
+    this.tableActions = [
+      {
+        permission: 'update',
+        icon: Icons.start,
+        click: () => this.daemonAction('start'),
+        name: this.actionLabels.START,
+        disable: () => this.actionDisabled('start')
+      },
+      {
+        permission: 'update',
+        icon: Icons.stop,
+        click: () => this.daemonAction('stop'),
+        name: this.actionLabels.STOP,
+        disable: () => this.actionDisabled('stop')
+      },
+      {
+        permission: 'update',
+        icon: Icons.restart,
+        click: () => this.daemonAction('restart'),
+        name: this.actionLabels.RESTART,
+        disable: () => this.actionDisabled('restart')
+      },
+      {
+        permission: 'update',
+        icon: Icons.deploy,
+        click: () => this.daemonAction('redeploy'),
```

Comment:
I thought it was self-explanatory both restart and redeploy. Restart is stop and start I'd say, and redeploy recreates the systemd unit files, stops and restarts the service again. Reconfig I haven't checked since I thought it would increase the complexity of this pr rn